crates/core/src/timeline/lsm.rs

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+use crate::config::HudiConfigs;
+use crate::metadata::HUDI_METADATA_DIR;
+use crate::storage::Storage;
+use crate::timeline::instant::Instant;
+use crate::timeline::selector::TimelineSelector;
+use crate::Result;
+use log::debug;
+use std::sync::Arc;
+
+/// LSM timeline for table version 8.
+/// For now, falls back to reading from .hoodie/ directory like active timeline.
+/// TODO: Add LSM timeline history reading from .hoodie/timeline/ directory
+#[derive(Debug, Clone)]
+pub struct LsmTimeline {
+    storage: Arc<Storage>,
+}
+
+impl LsmTimeline {
```

Comment:
We can keep LSM tree related logic in here. As commented above, LSM tree is internal to `TimelineLoader::LayoutTwoActive, LayoutTwoCompacted`, so just those 2 loaders can make use of the logic here. Btw, the correct name is LSM tree, so let's keep it in full like `lsm_tree.rs` and `LSMTree` for other applicable names ?